### PR TITLE
Greeting page was created.

### PR DESCRIPTION
**Pull request by eliasrvt**
## Summary

- A simple controller to say Hello was created.

## Trello Card

https://trello.com/c/fyFO8029

### CODE REVIEW

**Review by sebas97-stack — APPROVED** [review]

**Review by sebas97-stack — APPROVED** [review]

**Review by sebas97-stack — DISMISSED** [review]

**Review by sebas97-stack — DISMISSED** [review]

File: src/main/java/Wolox/training/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
+package Wolox.training.models;
+
+import javax.persistence.*;
+
+@Entity
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (juandbc) [inline src/main/java/Wolox/training/models/Book.java]: When we use PostgreSQL, the best strategy is **sequence**, so Hibernate can use a sequence table optimized by PostgreSQL. It is like declaring the column as **SERIAL** in the DDL script.
```suggestion
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
```

File: src/main/java/Wolox/training/models/Book.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    @Column(nullable = false)
+    private String year;
+
+    @Column(nullable = false)
+    private Integer pages;
+
+    @Column(nullable = false)
+    private String isbn;
+
+    public Book(String title, String author, String gender, String image, String subtitle, String publisher, String year, Integer pages, String isbn) {
```
Comment (juandbc) [inline src/main/java/Wolox/training/models/Book.java:39]: Here is missing the default non args constructor that is required JPA
```suggestion

   public Book() {}

    public Book(String title, String author, String gender, String image, String subtitle, String publisher, String year, Integer pages, String isbn) {
```

File: src/main/java/Wolox/training/models/Book.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    @Column(nullable = false)
+    private String publisher;
+
+    @Column(nullable = false)
+    private String year;
+
+    @Column(nullable = false)
+    private Integer pages;
+
+    @Column(nullable = false)
```
Comment (juandbc) [inline src/main/java/Wolox/training/models/Book.java]: The isbn should be also unique

File: src/main/java/Wolox/training/models/Book.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
+package Wolox.training.models;
+
+import javax.persistence.*;
+
+@Entity
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
```
Comment (juandbc) [inline src/main/java/Wolox/training/models/Book.java]: When working with ORMs such as Hibernate, it is recommended to use the wrapper class for the type of the primary keys
```suggestion
    private Long id;
```

File: build.gradle [hunk 5]
```diff
@@ -19,6 +19,7 @@ dependencies {
 	implementation('org.springframework.boot:spring-boot-starter-web')
 	implementation group: 'org.postgresql', name: 'postgresql', version: '42.2.5'
```
Comment (juandbc) [inline build.gradle:20]: For unit tests, it is recommended to use an in-memory database such as H2, so the project does not depend on a database implementation.
```suggestion
	implementation group: 'org.postgresql', name: 'postgresql', version: '42.2.5'
	testImplementation 'com.h2database:h2:1.4.200'
```

**Review by juandbc — CHANGES_REQUESTED** [review]

**Review by juandbc — APPROVED** [review]

**Review by sebas97-stack — APPROVED** [review]

**Pull request merged**